### DISCUSSION TITLE
Ask user information after first login?

### DISCUSSION THREAD

**Discussion by GregoryWullimann** [nextauthjs/next-auth#6150] [Help]
I want users to insert some information after the sign up with OAuth.

The logic should be something like:
- User signup with Oauth (google, apple, etc.)
- After a successful sign up, user is redirected to `/auth/new-user`
- User fill a form with data (username, description, etc.)
- User complete the sign up procedure

If the user close the `/auth/new-user` page without filling the information, it should be logged out and the next time he logins he should be treated like a "non-existing" user. So the next time he logins, it should be redirected again to `/auth/new-user` and ask information.

An example of this flow can be found at [chess.com](https://chess.com/).

What would be the correct way to implement something like this?

**Comment by Hyodori04** [comment]
I think this logic should be handled in next-auth. Is there anyone who can developed this case with next-auth?
